src/Dialect/Mlir/DialectBuilder.cpp

```diff
@@ -942,4 +953,47 @@ FlatSymbolRefAttr LLVMBuilder::getOrInsertSymbolRef(ModuleOp module,
   return SymbolRefAttr::get(b.getContext(), funcName);
 }
 
+void LLVMBuilder::ifThenElse(
+    mlir::function_ref<Value(LLVMBuilder &createLLVM)> cond,
+    mlir::function_ref<void(LLVMBuilder &createLLVM)> thenFn,
+    mlir::function_ref<void(LLVMBuilder &createLLVM)> elseFn) const {
```

Comment:
Can you create a custom type using a "using"

Reply:
Done. Thanks!